File: db4oj.tests/src/com/db4o/db4ounit/common/cs/config/ServerConfigurationItemIntegrationTestCase.java
/* Copyright (C) 2009  Versant Inc.   http://www.db4o.com */
package com.db4o.db4ounit.common.cs.config;

import com.db4o.*;
import com.db4o.cs.*;
import com.db4o.cs.config.*;
import com.db4o.io.*;

import db4ounit.*;

public class ServerConfigurationItemIntegrationTestCase implements TestCase {

	public void test() {
		ServerConfiguration config = Db4oClientServer.newServerConfiguration();
		config.file().storage(new MemoryStorage());
		DummyConfigurationItem item = new DummyConfigurationItem();
		config.addConfigurationItem(item);
		ObjectServer server = Db4oClientServer.openServer(config, "", Db4oClientServer.ARBITRARY_PORT);
		item.verify(config, server);
		server.close();
	}

	private final class DummyConfigurationItem implements ServerConfigurationItem {
		private int _prepareCount = 0;
		private int _applyCount = 0;
		private ServerConfiguration _config;
		private ObjectServer _server;
		
		public void prepare(ServerConfiguration configuration) {
			_config = configuration;
			_prepareCount++;
		}

		public void apply(ObjectServer server) {
			_server = server;
			_applyCount++;
		}
		
		void verify(ServerConfiguration config, ObjectServer server) {
			Assert.areSame(config, _config);
			Assert.areSame(server, _server);
			Assert.areEqual(1, _prepareCount);
			Assert.areEqual(1, _applyCount);
		}
	}

}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/CS/Config/ServerConfigurationItemIntegrationTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using Db4oUnit;
using Db4objects.Db4o;
using Db4objects.Db4o.CS;
using Db4objects.Db4o.CS.Config;
using Db4objects.Db4o.IO;
using Db4objects.Db4o.Tests.Common.CS.Config;

namespace Db4objects.Db4o.Tests.Common.CS.Config
{
	public class ServerConfigurationItemIntegrationTestCase : ITestCase
	{
		public virtual void Test()
		{
			IServerConfiguration config = Db4oClientServer.NewServerConfiguration();
			config.File.Storage = new MemoryStorage();
			ServerConfigurationItemIntegrationTestCase.DummyConfigurationItem item = new ServerConfigurationItemIntegrationTestCase.DummyConfigurationItem
				(this);
			config.AddConfigurationItem(item);
			IObjectServer server = Db4oClientServer.OpenServer(config, string.Empty, Db4oClientServer
				.ArbitraryPort);
			item.Verify(config, server);
			server.Close();
		}

		private sealed class DummyConfigurationItem : IServerConfigurationItem
		{
			private int _prepareCount = 0;

			private int _applyCount = 0;

			private IServerConfiguration _config;

			private IObjectServer _server;

			public void Prepare(IServerConfiguration configuration)
			{
				this._config = configuration;
				this._prepareCount++;
			}

			public void Apply(IObjectServer server)
			{
				this._server = server;
				this._applyCount++;
			}

			internal void Verify(IServerConfiguration config, IObjectServer server)
			{
				Assert.AreSame(config, this._config);
				Assert.AreSame(server, this._server);
				Assert.AreEqual(1, this._prepareCount);
				Assert.AreEqual(1, this._applyCount);
			}

			internal DummyConfigurationItem(ServerConfigurationItemIntegrationTestCase _enclosing
				)
			{
				this._enclosing = _enclosing;
			}

			private readonly ServerConfigurationItemIntegrationTestCase _enclosing;
		}
	}
}
#endif // !SILVERLIGHT
